### PR TITLE
Feat: sysctl default improvements

### PR DESCRIPTION
**Pull request by mbologna**

### CODE REVIEW

File: salt/ufw/etc_ufw/sysctl.conf [hunk 1]
```diff
@@ -32,10 +56,42 @@ net/ipv4/conf/default/log_martians=0
 #net/ipv4/tcp_fin_timeout=30
 #net/ipv4/tcp_keepalive_intvl=1800
 
+# Ignore Directed pings
+#net/ipv4/icmp_echo_ignore_all = 1
```
Comment (moio) [inline salt/ufw/etc_ufw/sysctl.conf]: Is this commented out intentionally?

File: salt/ufw/etc_ufw/sysctl.conf [hunk 2]
```diff
@@ -32,10 +56,42 @@ net/ipv4/conf/default/log_martians=0
...
+
 # Uncomment this to turn off ipv6 autoconfiguration
 #net/ipv6/conf/default/autoconf=1
 #net/ipv6/conf/all/autoconf=1
 
 # Uncomment this to enable ipv6 privacy addressing
 #net/ipv6/conf/default/use_tempaddr=2
 #net/ipv6/conf/all/use_tempaddr=2
+
+# Performance Recommended Settings:
```
Comment (moio) [inline salt/ufw/etc_ufw/sysctl.conf]: Are those settings related to security? Why should they be preferred?

**Review by moio — CHANGES_REQUESTED** [review]

File: salt/ufw/etc_ufw/sysctl.conf (hunk 1 above)
Comment (mbologna) [reply salt/ufw/etc_ufw/sysctl.conf]: Yes, at line 49 we are allowing ICMP echo. However, if someone wants to blocka all echo should uncomment this line.

File: salt/ufw/etc_ufw/sysctl.conf (hunk 2 above)
Comment (mbologna) [reply salt/ufw/etc_ufw/sysctl.conf]: The following settings are not related to security: they are more directed towards performance: 
https://www.cyberciti.biz/faq/linux-tcp-tuning/

Do you think it's a good idea to separate these settings into another sysctl-sls related file?

File: salt/ufw/etc_ufw/sysctl.conf (hunk 1 above)
Comment (moio) [reply salt/ufw/etc_ufw/sysctl.conf]: Then please either remove this line or comment it inline.

File: salt/ufw/etc_ufw/sysctl.conf (hunk 2 above)
Comment (moio) [reply salt/ufw/etc_ufw/sysctl.conf]: No, a separate file is not really necessary.

It is more of a matter of titles - commit and PR titles. This PR has "security" in its title, so this change does not really belong here.

Please either:
 - change the PR title to something more generic (like "sysct default improvements") and distinguish security and performance settings in commit messages, or
 - create a separate PR for performance settings

Furthermore, do you have any indication as to why Jessie defaults are inadequate in this case, or any measurable benefit from these settings? I would not want to maintain a different set of settings than the defaults unless it is beneficial.

Comment (mbologna) [reply salt/ufw/etc_ufw/sysctl.conf]: This contains the same adjustments and claims to be a reference among tweakers: http://www.slideshare.net/AmazonWebServices/pfc306-performance-tuning-amazon-ec2-instances-aws-reinvent-2014

Comment (moio) [reply salt/ufw/etc_ufw/sysctl.conf]: This is 1) specific to AWS 2) "take only when prescribed from performance engineer" 3) "best before 2015" and most importantly 4) unlikely to get any benefit in this scenario.

I am not going to merge the performance part, please remove it from this PR.

Comment (mbologna) [reply salt/ufw/etc_ufw/sysctl.conf]: Performance part has been removed. Only the security part is still present.

**Review by moio — APPROVED** [review]

**Pull request merged**